protocols/kad/src/query/peers/closest.rs

```diff
@@ -175,10 +175,14 @@ impl ClosestPeersIter {
             },
         }
 
-        let num_closest = self.closest_peers.len();
-        let mut progress = false;
-
         // Incorporate the reported closer peers into the iterator.
+        //
+        // The iterator makes progress if:
+        //     1, the iterator did not yet accumulate enough closest peers.
+        //   OR
+        //     2, any of the new peers is closer to the target than any peer seen so far
+        //        (i.e. is the first entry after incorporated)
```

Comment:
```suggestion
        //        (i.e. is the first entry after being incorporated)
```

Not a native speaker. I would assume this is easier to understand.

Reply:
PR updated according to the comment